core/src/main/java/com/digitalpebble/stormcrawler/protocol/okhttp/HttpProtocol.java

```diff
@@ -248,20 +251,25 @@ public ProtocolResponse getProtocolOutput(String url, final Metadata metadata) t
                 String key = headers.name(i);
                 String value = headers.value(i);
 
-                if (key.equalsIgnoreCase(VERBATIM_REQUEST_KEY) || key.equalsIgnoreCase(VERBATIM_RESPONSE_KEY)) {
+                if (key.equals(ProtocolResponse.REQUEST_HEADERS_KEY)
+                        || key.equals(ProtocolResponse.RESPONSE_HEADERS_KEY)) {
                     value = new String(Base64.getDecoder().decode(value));
                 }
 
-                responsemetadata.addValue(key.toLowerCase(Locale.ROOT), value);
```

Comment:
@sebastian-nagel not normalising the keys to lowercase has serious implications on the  subsequent steps and is also incompatible with what the other implementation produces. What problem were you trying to solve here?

Reply:
Looks like a mistake - given that this was together with moving custom headers to constants (being lowercase).

Reply:
After a closer look: that's a regression. I see: HTTP header names are required in lowercase everywhere else. Sorry.

Reply:
no problems, I should have spotted it when merging. Thanks for confirming!

Reply:
Yeah, it would break some other things, eg. last-modified handling. Maybe add a comment?
Btw., the `source.buffer()` calls are deprecated, should be `getBuffer()` now.